test/test_quicktypes.jl

```diff
@@ -69,7 +69,7 @@ end
 @testset "Pack" begin
     x = Pack([Cat(:Tama, 1), Cat(:Pochi, 2)])
 
-    @set! x.animals[2].nlegs = 5
+    x = @set x.animals[2].nlegs = 5
```

Comment:
I think this part is not tested since I commented `test_quicktypes.jl` out in https://github.com/jw3126/Setfield.jl/pull/31#discussion_r194624674. I suggest to pull #36 first or cherry-pick it.

Reply:
Ah, so the mutability tests fail there with Julia 1.0.  I guess this PR should go first.

Reply:
So #37 is failing as expected.

Reply:
So it is failing because `IndexLens` does not support `Array` anymore.  I implemented a fix in #40.

Another (non-) solution is to use tuples in test_quicktypes.jl: #39